client/state/sites/actions.js

```diff
@@ -135,8 +135,9 @@ export function requestSite( siteFragment ) {
 		const result = doRequest( false ).catch( ( error ) => {
 			// if there is Jetpack JSON API module error, retry with force: 'wpcom'
 			if (
-				error?.status === 403 &&
-				error?.message === 'API calls to this blog have been disabled.'
+				( error?.status === 403 &&
+					error?.message === 'API calls to this blog have been disabled.' ) ||
+				( error?.status === 400 && error?.name === 'ApiNotFoundError' )
```

Comment:
I know it's been here long before this PR, but we shouldn't check the specific text of the `error.message`. It's intended only for humans, and in general case can be localized. Checking for `error?.name === 'UnauthorizedError'` would be better.

We could be even more liberal and retry whenever the `status` is `4xx`. The error case is rare and an extra `force=wpcom` request doesn't hurt.

Reply:
Thanks for your review @jsnajdr 

Good point 👍 I agree. The issue is a blocker for the Search plugin and I'll merge this for now and revisit this and see what the best way to do the retry logic here. Thanks again 💯